sdk/python/kfp/components/_structures.py

```diff
@@ -334,14 +335,38 @@ def _post_init(self) -> None:
             raise TypeError('Need at least one argument.')
 
 
+class GraphInputReference(ModelBase):
+    '''References the input of the graph (the scope is a single graph).'''
+    _serialized_names = {
+        'input_name': 'inputName',
+    }
+
+    def __init__(self,
+        input_name: str,
+        type: Optional[TypeSpecType] = None,    # Can be used to override the reference data type
```

Comment:
Maybe not in this PR but it seems like we have a duplicated def for `TypeSpecType` (also in kfp.dsl.types) It would be great if we can get rid of one of them.

Reply:
>It would be great if we can get rid of one of them.

Sure. I have it on my plan and I'll unify those when revamping the typing and type checking.